File: ulyanovsk/MVC/Controllers/DetailsViewController.m
//
//  DetailsViewController.m
//  ulyanovsk
//
//  Created by Ekaterina Kolesnikova on 13.02.16.
//  Copyright © 2016 Ekaterina Kolesnikova. All rights reserved.
//

#import "DetailsViewController.h"
#import "ulyanovsk-Swift.h"
#import "UIImageView+WebCache.h"

@interface DetailsViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UILabel *infoLabel;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *contentViewWidth;

@end
@implementation DetailsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.title = self.place.title;
    [self.imageView sd_setImageWithURL:[NSURL URLWithString:[NSString stringWithFormat:@"https://arcane-brook-20885.herokuapp.com/%@", self.place.image]]
                      placeholderImage:[UIImage imageNamed:@"placeholder"]];
    
    self.titleLabel.text = self.place.title;
    self.infoLabel.text = self.place.info;
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    self.contentViewWidth.constant = self.view.frame.size.width;
    [self.view layoutIfNeeded];
}

- (IBAction)showMap:(id)sender
{
}

- (IBAction)showReviews:(id)sender
{
}

@end

File: ulyanovsk/MVC/Controllers/DetailsViewController.swift
//
//  DetailsViewController.swift
//  ulyanovsk
//
//  Created by Kolesnikova Ekaterina on 28/01/16.
//  Copyright © 2016 Ekaterina Kolesnikova. All rights reserved.
//

import UIKit

class DetailsViewController: UITableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    // MARK: - Table view data source

    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 3
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let reuseIdentifier = indexPath.row == 0 ? "info_cell" : "next_action_cell"
        let cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier, forIndexPath: indexPath)
        return cell
    }
    
    override func tableView(tableView: UITableView, willDisplayCell cell: UITableViewCell, forRowAtIndexPath indexPath: NSIndexPath) {
        //Для ячейки с описанием
        if indexPath.row == 0 {
            return
        }
        //Для других ячеек
        let label = cell.viewWithTag(1) as! UILabel;
        switch indexPath.row {
        case 1: label.text = "Показать на карте"
        case 2: label.text = "Отзывы"
        default: break
        }
        
    }
    
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        tableView.deselectRowAtIndexPath(indexPath, animated: true)
        
    }
    
}
